data/lua/stun.lua

```diff
@@ -0,0 +1,72 @@
+
+local _ = wesnoth.textdomain "wesnoth-units"
+local on_event = wesnoth.require("on_event")
+local old_unit_status = wesnoth.interface.game_display.unit_status
+
+function wesnoth.interface.game_display.unit_status()
+    local u = wesnoth.interface.get_displayed_unit()
+    if not u then return {} end
+    local s = old_unit_status()
+
+    if u.status.stunned then
+        table.insert(s, wml.tag.element{
+            image = "misc/stunned-status-icon.png",
+            tooltip = _"stunned: This unit is stunned. It cannot enforce its Zone of Control."
+        })
+    end
+
+    return s
+end
+
+local function on_hit(weapon, opponent)
+    if not weapon:matches{special_id = 'stun'} then return end
+    if not opponent.zoc then return end
+    local text
+    if opponent.gender == "female" then
+        text = _ "stunned"
+    else
+        text = _ "female^stunned"
+    end
+	local color = stringx.join(',', {'196', '196', '128'})
```

Comment:
Indentation

Reply:
I have no idea what you're talking about, but if the fix-whitespace script doesn't deal with it I can't be bothered.

Reply:
You should be bothered, no point in having low-quality code if it's little effort to fix.

Reply:
If there was a visible indentation issue I'd be bothered, but it looks fine to me.

Reply:
It looks right in your editor, with the tab-stops set to 4 spaces. It looks wrong in Github's diff, and possibly other editors.

Reply:
Currently, there's only one .lua file that has any tabs in at all - WC's unittypedata.lua. I thought it would be complex to check for files that mix tabs and spaces, but a blanket "no tabs in .lua files" check would be easy. Will open a PR later.

Reply:
I don't see any issue on GitHub either.